modules/yaml/src/main/scala/pureconfig/module/yaml/package.scala

```diff
@@ -0,0 +1,151 @@
+package pureconfig.module
+
+import java.io.{ IOException, Reader }
+import java.nio.file.{ Files, Path }
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+import scala.util.Try
+import scala.util.control.NonFatal
+
+import com.typesafe.config.{ ConfigValue, ConfigValueFactory }
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
+import org.yaml.snakeyaml.error.{ Mark, MarkedYAMLException, YAMLException }
+import pureconfig.ConvertHelpers._
+import pureconfig._
+import pureconfig.error._
+import pureconfig.module.yaml.error.{ NonStringKeyFound, UnsupportedYamlType }
+
+package object yaml {
+
+  // Converts an object created by SnakeYAML to a Typesafe `ConfigValue`.
+  // (https://bitbucket.org/asomov/snakeyaml/wiki/Documentation#markdown-header-loading-yaml)
+  private[this] def yamlObjToConfigValue(obj: AnyRef): Either[ConfigReaderFailures, ConfigValue] = {
+
+    def aux(obj: AnyRef): Either[ConfigReaderFailures, AnyRef] = obj match {
+      case m: java.util.Map[Object @unchecked, Object @unchecked] =>
```

Comment:
Just curious: Why represent the types in the `Map` as `Object` rather than `AnyRef`? Also, why the `@unchecked` annotation.

Reply:
The `Object` references were copy-paste, changed them to `AnyRef` in 3db81a762adb60311dedf68dc885b0743278762d.

We need the `@unchecked` since the Scala compiler cannot ensure that the type params are really we ones we asked for in pattern matches, due to type erasure. We need this annotation so that the compiler doesn't yield a warning.